src/game/logic/object/update/aiupdate.h

```diff
@@ -56,21 +68,51 @@ enum LocomotorSetType
     LOCOMOTORSET_COUNT,
 };
 
+enum MoodMatrixAction
+{
+    MM_ACTION_IDLE = 0,
+    MM_ACTION_MOVE,
+    MM_ACTION_ATTACK,
+    MM_ACTION_ATTACKMOVE,
+};
+
 class AICommandInterface
 {
 public:
     virtual void AI_Do_Command(AICommandParms const *params) = 0;
 };
 
-class AIUpdateModuleData
+class AIUpdateModuleData : public UpdateModuleData
 {
 public:
+    virtual ~AIUpdateModuleData() override;
+    virtual bool Is_AI_Module_Data() const override;
+
     static void Parse_Locomotor_Set(INI *ini, void *formal, void *store, const void *user_data);
+
+private:
+    std::map<LocomotorSetType, LocomotorTemplate const *> m_locomotorTemplates;
+    TurretAIData *m_turretData[2];
+    unsigned int m_moodAttackCheckRate;
+    bool m_forbidPlayerCommands;
+    bool m_turretsLinked;
+    int m_autoAcquireEnemiesWhenIdle;
+    friend class AIUpdateInterface;

```

Comment:
Could you put friends always at the very top of the class body? I think friends are too important to be written so far behind, being barely visible here next to the members.